packages/format-library/src/link/modal.native.js

```diff
@@ -0,0 +1,155 @@
+
+import React from 'react';
+import { Button, Switch, Text, TextInput, View } from 'react-native';
+import Modal from 'react-native-modal';
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import { URLInput } from '@wordpress/editor';
+import { prependHTTP, safeDecodeURI, filterURLForDisplay } from '@wordpress/url';
+import {
+	withSpokenMessages,
+} from '@wordpress/components';
+import {
+	create,
+	insert,
+	isCollapsed,
+	applyFormat,
+	getTextContent,
+	remove,
+	slice,
+} from '@wordpress/rich-text';
+
+import { createLinkFormat, isValidHref } from './utils';
+
+import styles from './modal.scss';
+
+class ModalLinkUI extends Component {
+	constructor( props ) {
+		super( ...arguments );
+
+		this.submitLink = this.submitLink.bind( this );
+		this.onChangeInputValue = this.onChangeInputValue.bind( this );
+		this.onChangeText = this.onChangeText.bind( this );
+		this.onChangeOpensInNewWindow = this.onChangeOpensInNewWindow.bind( this );
+
+		this.state = {
+			inputValue: '',
+			text: getTextContent( slice( props.value ) ),
+			opensInNewWindow: false,
+		};
+	}
+
+	onChangeInputValue( inputValue ) {
+		this.setState( { inputValue } );
+	}
+
+	onChangeText( text ) {
+		this.setState( { text } );
+	}
+
+	onChangeOpensInNewWindow( opensInNewWindow ) {
+		this.setState( { opensInNewWindow } );
+	}
+
+	submitLink() {
+		const { isActive, value, onChange, speak } = this.props;
+		const { inputValue, opensInNewWindow, text } = this.state;
+		const url = prependHTTP( inputValue );
+		const format = createLinkFormat( {
+			url,
+			opensInNewWindow,
+			text,
+		} );
+
+		if ( isCollapsed( value ) && ! isActive ) {
+			const toInsert = applyFormat( create( { text } ), format, 0, text.length );
+			onChange( insert( value, toInsert ) );
+		} else if ( text !== getTextContent( slice( value ) ) ) {
+			const toInsert = applyFormat( create( { text } ), format, 0, text.length );
+			onChange( insert( value, toInsert, value.start, value.end ) );
+		} else {
+			onChange( applyFormat( value, format ) );
+		}
+
+		if ( ! isValidHref( url ) ) {
+			speak( __( 'Warning: the link has been inserted but may have errors. Please test it.' ), 'assertive' );
+		} else if ( isActive ) {
+			speak( __( 'Link edited.' ), 'assertive' );
+		} else {
+			speak( __( 'Link inserted' ), 'assertive' );
+		}
+
+		this.props.onClose();
+	}
+
+	render() {
+		const { isVisible } = this.props;
+
+		return (
+			<Modal
+				isVisible={ isVisible }
+				style={ styles.bottomModal }
+				animationInTiming={ 500 }
+				animationOutTiming={ 500 }
+				backdropTransitionInTiming={ 500 }
+				backdropTransitionOutTiming={ 500 }
+				onBackdropPress={ this.props.onClose }
+				onSwipe={ this.props.onClose }
+				swipeDirection="down"
+			>
+				<View style={ { ...styles.content, borderColor: 'rgba(0, 0, 0, 0.1)' } }>
+					<View style={ styles.head }>
+						<Button
```

Comment:
We can't reuse `Button` from `@wordpress/components` as they currently have hardcoded styles and it's outside of the scope of this PR to refactor it imo.
Let's use Button from `react-native` in the meantime even though we might not be able to style them as much as we'd like (wrap text)